### PR TITLE
Add back snapshot build restriction for match and qstr functions

### PR DESCRIPTION
**Pull request by carlosdelest**
This reverts https://github.com/elastic/elasticsearch/pull/114793

match() and qstr() will be back behind snapshot builds on 8.x branches until using them after LIMIT has been addressed, either in the functions themselves or in Kibana side.

### CODE REVIEW

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by carlosdelest** [comment]
> are we not reverting on 8.17 & 8.18 so its easier to test and develop with?

@benwtrent the idea is to fix the LIMIT limitations for 8.17. We could of course be cautious and remove it from the 8.x branch - we're just thinking ahead to avoid another change for enabling this on 8.x

**Review by ChrisHegarty — CHANGES_REQUESTED** [review]
Please await further discussion on this revert.

**Review by ChrisHegarty — APPROVED** [review]

File: x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4 [hunk 1]
```diff
@@ -305,8 +305,8 @@ mode PROJECT_MODE;
 PROJECT_PIPE : PIPE -> type(PIPE), popMode;
 PROJECT_DOT: DOT -> type(DOT);
 PROJECT_COMMA : COMMA -> type(COMMA);
-PROJECT_PARAM : {this.isDevVersion()}? PARAM -> type(PARAM);
```
[the remark this answers is not in the document]
Comment (carlosdelest) [reply x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4]: Thanks for the catch! No they're not - fixed

**Pull request merged**